provider/image.go

```diff
@@ -428,6 +438,43 @@ func getCredentials() (map[string]clitypes.AuthConfig, error) {
 	return auths, nil
 }
 
+// Because the authConfigs provided by the host include the `https://` prefix in the map keys, `getRegistryAddrForAuth`
+// ensures we return a registry address that includes the `https://` scheme.
+// While this prefix is not needed for program-provided auth, it is valid regardless, so adding it by default
+// keeps the special case handling to a minimum.
+func getRegistryAddrForAuth(serverName, imgName string) (string, error) {
+	var serverAddr string
+	if serverName == "docker.io" {
+		// if it's dockerhub, we special case it so host config can find the correct registry
+		return "https://index.docker.io/v1/", nil
+	}
+
+	if serverName == "" {
+		// if there is no servername in the registry input, we attempt to build it from the fully qualified image name.
+		addr, _, found := strings.Cut(imgName, "/")
+		if !found {
+			return "", errors.Errorf("image name must be fully qualified: %s", imgName)
+		}
+		if addr == "docker.io" {
+			return "https://index.docker.io/v1/", nil
+		}
+		// we need the full server address for the lookup
+		serverAddr = "https://" + addr
```

Comment:
Is it ever valid for `addr` to start with `https://` or `http://` or some other url prefix? Should we error if `addr` starts with `http(s?)://`, or should we trim it?

Especially considering L466-471.

Reply:
The `https://` is being added for the purpose of looking up the registry from the host docker config and we should not trim it.

As far as I know, with the special snowflake exception of `docker.io` there is no other url prefix that is valid here. The `else` block is a best effort attempt to make sure we can always look up the registry from the host config if necessary, since including it is always valid.